viewer/packages/sector-parser/src/GltfSectorParser.ts

```diff
@@ -0,0 +1,165 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+import * as THREE from 'three';
+import { MeshGPUInstancing, InstancedMesh } from '@gltf-transform/extensions';
+import { Node, NodeIO, Document, Primitive } from '@gltf-transform/extensions/node_modules/@gltf-transform/core';
+import {
+  setBoxGeometry,
+  setConeGeometry,
+  setNutGeometry,
+  setQuadGeometry,
+  setTorusGeometry,
+  setTrapeziumGeometry
+} from './primitiveGeometries';
+
+export enum RevealGeometryCollectionType {
+  BoxCollection,
+  CircleCollection,
+  ConeCollection,
+  EccentricConeCollection,
+  EllipsoidSegmentCollection,
+  GeneralCylinderCollection,
+  GeneralRingCollection,
+  QuadCollection,
+  TorusSegmentCollection,
+  TrapeziumCollection,
+  NutCollection,
+  TriangleMesh,
+  InstanceMesh
+}
+
+export default class GltfSectorParser {
+  private readonly _io: NodeIO;
+  constructor() {
+    this._io = new NodeIO();
+    this._io.registerExtensions([MeshGPUInstancing]);
+  }
+
+  public parseSector(data: ArrayBuffer) {
+    const document: Document = this._io.readBinary(data);
+
+    const defaultScene = document.getRoot().getDefaultScene();
+    const buffs: [RevealGeometryCollectionType, THREE.BufferGeometry][] = [];
+    defaultScene!.traverse((node: Node) => {
+      const res = this.processNode(node);
+      if (res) buffs.push(res);
+    });
+
+    return buffs;
+  }
+
+  private processNode(node: Node): [RevealGeometryCollectionType, THREE.BufferGeometry] | null {
+    const instanced = node.getExtension<InstancedMesh>(MeshGPUInstancing.EXTENSION_NAME)!;
+    const mesh = node.getMesh()!;
+
+    if (!instanced && !mesh) return null;
```

Comment:
When does this happen? Is this ignoring stuff we don't support?

Reply:
In practice, never because the backend will not produce empty nodes. It is just a check (actually it should probably be an assert).